test/functional/eval/execute_spec.lua

```diff
@@ -75,17 +75,59 @@ describe('execute()', function()
   end)
 
   it('silences command run inside', function()
-    local screen = Screen.new(40, 5)
-    screen:attach()
-    screen:set_default_attr_ids( {[0] = {bold=true, foreground=255}} )
-    feed(':let g:mes = execute("echon 42")<CR>')
-    screen:expect([[
-    ^                                        |
-    {0:~                                       }|
-    {0:~                                       }|
-    {0:~                                       }|
-    :let g:mes = execute("echon 42")        |
-    ]])
-    eq('42', eval('g:mes'))
+  end)
+
+  describe('{silent} argument', function()
+    it('captures & displays output for ""', function()
+      local screen = Screen.new(40, 5)
+      screen:attach()
+      command('let g:mes = execute("echon 42", "")')
+      screen:expect([[
+      ^                                        |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      42                                      |
+      ]])
+      eq('42', eval('g:mes'))
+    end)
+
+    it('captures but does not display output for "silent"', function()
+      local screen = Screen.new(40, 5)
+      screen:attach()
+      command('let g:mes = execute("echon 42")')
+      screen:expect([[
+      ^                                        |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+                                              |
+      ]])
+      eq('42', eval('g:mes'))
+
+      command('let g:mes = execute("echon 13", "silent")')
+      screen:expect([[
+      ^                                        |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+                                              |
+      ]])
+      eq('13', eval('g:mes'))
+    end)
+
+    it('suppresses errors for "silent!"', function()
+      eq(0, exc_exec('let g:mes = execute(0.0, "silent!")'))
+      eq('', eval('g:mes'))
+    end)
+
+    it('propagates errors for "" and "silent"', function()
+      local ret
+      ret = exc_exec('call execute(0.0, "")')
+      eq('Vim(call):E806: using Float as a String', ret)
+
+      ret = exc_exec('call execute(v:_null_dict, "silent")')
```

Comment:
Isn't this a validation error? So it's not testing the codepath where `f_execute()` actually runs commands that themselves produce errors.